packages/ffe-searchable-dropdown-react/src/List.js

```diff
@@ -0,0 +1,235 @@
+import {
+    CellMeasurer,
+    CellMeasurerCache,
+    List as VirtualizedList,
+    AutoSizer,
+} from 'react-virtualized';
+import React from 'react';
+import { Scrollbars } from 'react-custom-scrollbars';
+import { Paragraph } from '@sb1/ffe-core-react';
+import {
+    arrayOf,
+    any,
+    bool,
+    func,
+    number,
+    object,
+    oneOf,
+    shape,
+    string,
+} from 'prop-types';
+
+import { getNotMatchText, locales } from './translations';
+import ListItemContainer from './ListItemContainer';
+import { stateChangeTypes } from './reducer';
+
+const NoMatch = ({ noMatch, noMatchMessageId, listToRender }) => (
+    <div>
+        {noMatch.text ? (
+            <div className="ffe-searchable-dropdown__no-match">
+                <Paragraph id={noMatchMessageId}>{noMatch.text}</Paragraph>
+            </div>
+        ) : (
+            listToRender.length === 0 && (
+                <Paragraph
+                    id={noMatchMessageId}
+                    className="ffe-screenreader-only"
+                >
+                    {getNotMatchText()}
+                </Paragraph>
+            )
+        )}
+    </div>
+);
+
+NoMatch.propTypes = {
+    noMatch: shape({
+        text: string,
+        dropdownList: arrayOf(object),
+    }),
+    noMatchMessageId: string,
+    listToRender: arrayOf(object).isRequired,
+};
+
+export default class List extends React.PureComponent {
+    state = {
+        optionHeight: 40,
+    };
+
+    constructor(props, context) {
+        super(props, context);
+
+        this._cache = new CellMeasurerCache({
+            fixedWidth: true,
+            minHeight: 10,
+        });
+
+        this.rowRenderer = this.rowRenderer.bind(this);
+        this.handleScroll = this.handleScroll.bind(this);
+
+        this.listRef = React.createRef();
+    }
+
+    handleScroll = event => {
+        const { scrollTop, scrollLeft } = event.target;
+        const { Grid } = this.listRef.current;
+        Grid.handleScrollEvent({ scrollTop, scrollLeft });
+    };
+
+    componentDidMount() {
+        requestAnimationFrame(() => {
+            const firstOption =
+                this.listRef.current &&
+                this.listRef.current.Grid &&
+                this.listRef.current.Grid._scrollingContainer &&
+                this.listRef.current.Grid._scrollingContainer.querySelector(
+                    '[role="option"]',
+                );
+
+            if (firstOption && firstOption.offsetHeight) {
```

Comment:
firstOption?.offsetHeight

Reply:
Prata med Torkjel og kom fram til at vi lar det være sånn her selv om vi ikke lenger trenger å støtte IE11. Det kan jo være at f.eks. Fremtind eller noe tar det her i bruk og at de støtter IE11 :)

Reply:
Jag er nok  ikke enig :) 
Alle bruker vell babel. Vi har brukt denne featuren massor hos og det var innen ie11 ble droppet. 


https://babeljs.io/docs/en/babel-plugin-proposal-optional-chaining

Reply:
Så dersom man har babel, så vil dette funke for IE11 også?

Reply:
Hvis man bruker denne så kommer den per default og man trenger ikke installera noe separat. Lurer på om man kan bruke serskillt mye av designbiblioteket i ie11 uten babel? Kanske mer en jag tror men idag er vell ikke noen som lager frontend uten babel?
Optional Chaining  er ellers stage 4  hos TC39 vilket betyr att den ikke lenger bare er ett forslag utan noe som kommer bli.   
Ser dere forsatt noen grund til ikke bruke det skall dere få lov att ikke bruke det alltså men jag liker syntaxen spm hjelper på lesbarheten plus att det da er safe og bruke mener jag da.

https://caniuse.com/mdn-javascript_operators_optional_chaining

https://babeljs.io/docs/en/babel-preset-env

Reply:
Enig i at det vil funke for de fleste, men et designsystem burde forvente minst mulig av slikt oppsett syns jeg. Vi kan jo la Mr. Designsystem @kwltrs avgjøre :smile:

Reply:
Vi bygger jo react-pakkene med babel og pakker inn commonjs- og es-moduler i npm-pakkene: https://github.com/SpareBank1/designsystem/blob/3d23895f14897615d4bf2a39c62b7a801acefb69/packages/ffe-searchable-dropdown-react/package.json#L23-L24
Og vi bruker @babel/preset-env: https://github.com/SpareBank1/designsystem/blob/develop/babel.config.js - den inkluderer IE11 som target. Så som konsument av en ffe-pakke burde man klare seg uten babel.

Reply:
Da har jeg pushet endringene med optional chaining. :blush: